src/Parallel/Algorithm.hpp

```diff
@@ -373,6 +380,36 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
   // @}
 
  private:
+  template <typename ThisVariant, typename... Variants, typename... Args>
+  void invoke_component_pup_with_current_box_impl(
+      PUP::er& p, boost::variant<Variants...>& box,
+      const gsl::not_null<int*> iter,
+      const gsl::not_null<bool*> already_visited) noexcept {
+    // void cast to avoid compiler warnings about the unused variable in the
+    // false branch of the constexpr
+    (void)already_visited;
+    if constexpr (detail::is_pup_callable_v<
+                      ParallelComponent, decltype(p), ThisVariant&,
+                      decltype(*global_cache_)&, array_index&>) {
+      if (box.which() == *iter and not *already_visited) {
+        ParallelComponent::pup(p, boost::get<ThisVariant>(box), phase_,
```

Comment:
pass box with `std::as_const` and add comment that if a non-const box is needed then the `Algorithm::pup` function needs to call this before  pupping `box_` when `p.isPacking()`